tests/integration/suite/test_istio.py

```diff
@@ -66,6 +66,48 @@ def test_destination_rule(admin_pc):
     client.delete(ns)
 
 
+# consistentHash has an "anyOf" openAPI validation on it,
+# and our types were passing multiple options which failed.
+# This test ensures you can pass a single option.
+# See: https://github.com/rancher/rancher/issues/25515
+def test_destination_rule_on_cookie(admin_pc, remove_resource):
+    client = admin_pc.client
+    ns = admin_pc.cluster.client.create_namespace(
+        name=random_str(),
+        projectId=admin_pc.project.id)
+    remove_resource(ns)
+    name = random_str()
+    cookie_name = name + "_cookie"
+    client.create_destinationRule(
+        name=name,
+        namespaceId=ns.id,
+        host="test",
+        subsets=[{
+            "name": "v1",
+            "labels": {
+                "version": "v1",
+            }
+        }],
+        trafficPolicy={
+            "loadBalancer": {
+                "consistentHash": {
+                    "httpCookie": {
+                        "ttl": "0s",
+                        "name": cookie_name,
+                    }
+                }
+            }
+        }
+    ),
```

Comment:
nit: remove_resource on destinationRule too, although it should be fine since ns will be deleted

Reply:
fixed